From 10c222773f6af391bb1225f42a6bd5e773afb575 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 04 Jun 2021 12:03:56 +0300
Subject: [PATCH] [SW-0] gaudi: security: improve hbw handover

To improve the BTL settings replacement w/o creating an opportunity
window, move the RR_0 setting to the end of the preboot_secure_hbw()
function.

Change-Id: I9080a095b40481bd3e2f7c239bccaf8e941f31e6
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 5ca44ea..b2aefaf 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1504,9 +1504,6 @@
 
 static void preboot_secure_hbw(void)
 {
-	/* HBM RR 0 - 32GB */
-	preboot_secure_hbw_rr(HBW_RR_HBM, HBM_RR_BASE, HBM_RR_MASK);
-
 	/* SRAM RR 0x7F_F000_0000 - 0x7F_F140_0000 (20MB) */
 	preboot_secure_hbw_sram();
 
@@ -1531,6 +1528,10 @@
 	/* PCIe FW SRAM RR 0x7F_FBFF_0000 - 0x7F_FBFF_8000 (32KB) */
 	preboot_secure_hbw_rr(HBW_RR_PCIE_FW, 0x7FFBFF0000, 0x3FFFFFFF8000);
 
+	/* HBM RR 0 - 32GB */
+	/* Keep RR_0 last, as it replaces the BTL settings */
+	preboot_secure_hbw_rr(HBW_RR_HBM, HBM_RR_BASE, HBM_RR_MASK);
+
 	/*
 	 * PCIe IF HBW RRs are for direct access from PCIe IF to PSOC domain:
 	 * STM, SPI Flash, Scratchpad SRAM, PCIe FW SRAM.
